src/build/buildExecutor.ts

```diff
@@ -1,20 +1,24 @@
 import * as fs from 'fs-extra';
+import * as path from 'path';
 import extensionConfig from '../config';
 import { PlatformInformation } from '../common/platformInformation';
 import { ChildProcess } from 'child_process';
 import { Package, AbsolutePathPackage } from '../dependency/package';
-import { DocfxRestoreCanceled, DocfxRestoreFailed, DocfxRestoreSucceeded, DocfxBuildCanceled, DocfxBuildSucceeded, DocfxBuildFailed, DocfxBuildStarted, DocfxRestoreStarted } from '../common/loggingEvents';
+import { DocfxBuildStarted, DocfxRestoreStarted, DocfxBuildCompleted, DocfxRestoreCompleted } from '../common/loggingEvents';
 import { EnvironmentController } from '../common/environmentController';
 import { EventStream } from '../common/eventStream';
 import { executeDocfx } from '../utils/childProcessUtils';
-import { basicAuth } from '../utils/utils';
+import { basicAuth, getDurationInSeconds } from '../utils/utils';
 import { ExtensionContext } from '../extensionContext';
+import { DocfxExecutionResult, BuildResult } from './buildResult';
+import { BuildInput } from './buildInput';
+import { OUTPUT_FOLDER_NAME } from '../shared';
 
 export class BuildExecutor {
     private cwd: string;
     private binary: string;
     private runningChildProcess: ChildProcess;
-    private static skipRestore: boolean;
+    private static skipRestore: boolean = false;

```

Comment:
No, there is no default value for boolean variable(default to be undefined). since it will be compiled to js, which does have type

---
In reply to: [380566195](https://github.com/docascode/vscode-docs-build/pull/44#discussion_r380566195) [](ancestors = 380566195)